src/liballoc/rc.rs

```diff
@@ -1172,6 +1177,8 @@ impl<T: ?Sized + PartialEq> RcEqIdent<T> for Rc<T> {
 /// store large values, that are slow to clone, but also heavy to check for equality, causing this
 /// cost to pay off more easily. It's also more likely to have two `Rc` clones, that point to
 /// the same value, than two `&T`s.
+///
+/// We can only do this when `T: Eq` as a `PartialEq` might be deliberately irreflexive.
```

Comment:
This is what I concluded after some thinking. It is also in line with [this comment](https://github.com/rust-lang/rust/pull/56550#issuecomment-446386849) by @alexcrichton:

> According to Eq's own documentation it signifies the reflexive property (a == a) which I believe means that this is a semantically correct implementation.

So, seems like just a missing comment to me.